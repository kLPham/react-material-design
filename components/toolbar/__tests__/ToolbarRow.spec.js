import React from 'react';
import { shallow } from 'enzyme';
import ToolbarRow from '../toolbarRow';

describe('<ToolbarRow />', () => {
    it('should render', () => {
        const toolbarRow = shallow(
            <ToolbarRow />,
    );
        expect(toolbarRow).toMatchSnapshot();
    });
    it('should render with a title', () => {
        const title = 'This is my Title';
        const toolbarRow = shallow(
            <ToolbarRow title={title} />,
      );
        expect(toolbarRow).toMatchSnapshot();
    });
    it('should render with leftElements', () => {
        const icon = <i className="material-icons">cloud</i>;
        const toolbarRow = shallow(
            <ToolbarRow leftElements={icon} />,
      );
        expect(toolbarRow).toMatchSnapshot();
    });
    it('should render with centerElements', () => {
        const icon = <i className="material-icons">cloud</i>;
        const toolbarRow = shallow(
            <ToolbarRow centerElements={icon} />,
      );
        expect(toolbarRow).toMatchSnapshot();
    });
    it('should render with rightElements', () => {
        const icon = <i className="material-icons">cloud</i>;
        const toolbarRow = shallow(
            <ToolbarRow rightElements={icon} />,
      );
        expect(toolbarRow).toMatchSnapshot();
    });
});
