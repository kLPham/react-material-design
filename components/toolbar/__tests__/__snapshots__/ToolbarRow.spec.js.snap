// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ToolbarRow /> should render 1`] = `
<div
  className="mdc-toolbar__row"
>
  <section
    className="mdc-toolbar__section mdc-toolbar__section--align-start"
  />
</div>
`;

exports[`<ToolbarRow /> should render with a title 1`] = `
<div
  className="mdc-toolbar__row"
>
  <section
    className="mdc-toolbar__section mdc-toolbar__section--align-start"
  >
    <span
      className="mdc-toolbar__title"
    >
      This is my Title
    </span>
  </section>
</div>
`;

exports[`<ToolbarRow /> should render with centerElements 1`] = `
<div
  className="mdc-toolbar__row"
>
  <section
    className="mdc-toolbar__section mdc-toolbar__section--align-start"
  />
  <section
    className="mdc-toolbar__section"
  >
    <i
      className="material-icons"
    >
      cloud
    </i>
  </section>
</div>
`;

exports[`<ToolbarRow /> should render with leftElements 1`] = `
<div
  className="mdc-toolbar__row"
>
  <section
    className="mdc-toolbar__section mdc-toolbar__section--align-start"
  >
    <i
      className="material-icons"
    >
      cloud
    </i>
  </section>
</div>
`;

exports[`<ToolbarRow /> should render with rightElements 1`] = `
<div
  className="mdc-toolbar__row"
>
  <section
    className="mdc-toolbar__section mdc-toolbar__section--align-start"
  />
  <section
    className="mdc-toolbar__section mdc-toolbar__section--align-end"
  >
    <i
      className="material-icons"
    >
      cloud
    </i>
  </section>
</div>
`;
